packages/lsp/src/connection.ts

```diff
@@ -268,16 +268,25 @@ export class LSPConnection extends LspWsConnection implements ILSPConnection {
     return this._serverInitialized;
   }
 
+  /**
+   * A signal emitted when the connection is disposed.
+   */
+  get disposed(): ISignal<this, void> {
+    return this._disposed;
+  }
+
```

Comment:
You should move this to the parent class (where the attribute is created) and switch `this._disposed` to a private attribute.